lib/matplotlib/_auto_placement.py

```diff
@@ -0,0 +1,87 @@
+"""
+Utility functions to find the best place in a plot for a rectangular box,
+e.g. a legend or an annotation.
+"""
+
+import numpy as np
+
+from matplotlib.lines import Line2D
+from matplotlib.patches import Patch, Rectangle
+from matplotlib.transforms import Bbox
+
+def find_best_position(ax, width, height, renderer, consider):
```

Comment:
Wondering why this was moved out of the module? Was it because we were going to use this in another place? Also, the contributing guide suggests adding docstrings for public methods, so it might be a good idea to add a docstring here.

Reply:
Yes, the point of this is that we're factoring this behaviour out to use it somewhere else.  And sure, I'll add a docstring